### PR TITLE
new(tests): EOF - EIP-4750: Stack  validation in CALLF

### PR DESCRIPTION
**Pull request by shemnon**
## 🗒️ Description

Add a test to cover cases where stack would not overflow but CALLF stack
reservation rules would cause a revert.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py [hunk 1]
```diff
@@ -252,3 +257,102 @@ def test_callf_with_inputs_stack_overflow(
...
+                ),
+            }
+        )
+    }
+
+    tx = Transaction(
+        to=contract_address,
+        gas_limit=10_000_000,
+        gas_price=10,
+        protected=False,
```
Comment (pdobacz) [inline tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py:355]: gas_price and protected can be left out

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py [hunk 2]
```diff
@@ -252,3 +257,102 @@ def test_callf_with_inputs_stack_overflow(
         ),
         container_post=Account(storage={slot_code_worked: 0}),
     )
+
+
+@pytest.mark.parametrize(
+    ("stack_height", "inputs", "outputs", "failure"),
+    (
+        pytest.param(1020, 1, 3, False, id="1020"),
+        pytest.param(1021, 1, 3, True, id="1021"),
```
Comment (pdobacz) [inline tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py:266]:
```suggestion
        pytest.param(1020, 1, 3, False, id="success"),
        pytest.param(1021, 1, 3, True, id="failure"),
```
if we just have these two now (do I understand correctly, the intention is that `1022` would not validate anymore? Or are there other combinations of params which would validate and make sense to test?

**Review by pdobacz — APPROVED** [review]
LGTM, minor comments. Do we need to also update `eof_tracker.md`? I'm not sure but maybe `Max stack size (1024) in CALLF-ed function` was covering this case, please double check though, I didn't analyze it

File: tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py (hunk 2 above)
Comment (shemnon) [reply tests/osaka/eip7692_eof_v1/eip4750_functions/test_callf_execution.py]: The essential test is pass below, and fail above.  I was considering messing with the input/output levels, but determined it was not essential to the test.  I took input and output of the params, and just have stack size and wether we expect failure.  Seemed to communicate intent better than embedding what a failure is into the text code.

**Comment by shemnon** [comment]
Added another tests to check CALLF when stack is at max ("rule #4") - by EIP this should revert, but there are ways it would succeed without the check, so we do this.

**Review by pdobacz — APPROVED** [review]

**Review by marioevz — APPROVED** [review]
LGTM, thanks!
